src/main/java/com/conveyal/r5/profile/Path.java

```diff
@@ -24,13 +27,43 @@
     // It does effectively allow you to leave out the boardStopPositions and alightStopPositions, but a subclass could do that too.
     public int[] patterns;
     public int[] boardStops;
+    // Note that including stop arrival times would imply a distinct path for each departure minute.
     public int[] alightStops;
     public int[] alightTimes;
     public int[] trips;
     public int[] boardStopPositions;
     public int[] alightStopPositions;
+    public StreetMode accessMode;
+    /**
+     * Used only in propagation for writing paths for Taui outputs
+     * TODO update egress mode outside of path (e.g. in wrapper or array parallel to perIterationTravelTimes in
+     *  propagater)
+     */
+    public StreetMode egressMode;
+
+    // Additional characteristics (not keys for hashing)
+    public int inVehicleTime;
+    public int waitTime;
     public final int length;
 
+    /**
+     * Copy a path and add the specified egress mode
+     */
+    public Path(Path input, StreetMode egressMode) {
```

Comment:
This might be easier to maintain over the long term if it is a static method that performs a clone() and then sets egressMode.